p2p/server.go

```diff
@@ -728,39 +694,74 @@ func (srv *Server) run(dialstate dialer) {
 		}
 	}
 
-	isValNode := func(ID enode.ID) bool {
-		if _, ok := valNodes[ID]; ok {
-			return true
+	addStatic := func(n *enode.Node, label string) {
+		if _, ok := static[n.ID()]; !ok {
+			static[n.ID()] = make(map[string]bool)
+		}
+		static[n.ID()][label] = true
+
+		// If already connected, updated val peer counters and set the validatorConn flag in the connection
+		if p, ok := peers[n.ID()]; ok {
+			if p.StaticNodeLabels == nil {
```

Comment:
What happens if p.StaticNodeLabels is not nil, would we not update the peer's labels with the new label then?

Reply:
The intent is to set the peer's stat node labels to a *pointer* to `static`'s label set.